PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Tests/ServicesTests/Appointment/AppointmentServiceTests.cs

```diff
@@ -0,0 +1,164 @@
+        };
+
+        _doctorRepoMock
+            .Setup(repo => repo.GetDoctorByFullName("John", "Doe"))
+            .ReturnsAsync(doctor);
+
+        _doctorRepoMock
+            .Setup(repo => repo.IsDoctorOnWork(It.IsAny<DateOnly>(), It.IsAny<TimeSpan>(), "doctor123"))
+            .ReturnsAsync(true);
+
+        _appointmentRepoMock
+            .Setup(repo => repo.IsDoctorHaveAppointment(It.IsAny<DateOnly>(), It.IsAny<TimeSpan>(), "doctor123"))
+            .ReturnsAsync(false);
+
+        _mapperMock
+            .Setup(mapper => mapper.Map<Appointment>(createDto))
+            .Returns(new Appointment { DoctorId = doctor.Id });
+
+        _appointmentRepoMock
+            .Setup(repo => repo.CreateAppointmentAsync(It.IsAny<Appointment>()))
+            .ReturnsAsync(true);
+
+        var result = await _appointmentService.CreateAppointment(createDto);
+        
+        Assert.True(result.Success);
+        Assert.True(result.Data);
+    }
+
+    [Fact]
+    public async Task CreateAppointment_ShouldReturnError_WhenDoctorNotFound()
+    {
+        var createDto = new CreateAppointmentDto
+        {
+            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(1)),
+            Time = TimeSpan.FromHours(10),
+            PatientId = "patient123",
+            DoctorFirstName = "Unknown",
+            DoctorLastName = "Doctor"
+        };
+
+        _doctorRepoMock
+            .Setup(repo => repo.GetDoctorByFullName("Unknown", "Doctor"))
+            .ReturnsAsync((AppUser)null);
+        
+        var result = await _appointmentService.CreateAppointment(createDto);
+        
+        Assert.False(result.Success);
+        Assert.Contains("Doctor not found", result.Errors);
+    }
+
+    [Fact]
+    public async Task GetAppointmentsBySpecialityId_ShouldReturnAppointments_WhenValidDataProvided()
+    {
+        var speciality = DoctorSpecialities.Surgeon;
+        var patientId = "patient123";
+
+        var appointments = new List<Appointment>
+        {
+            new Appointment { Date = DateOnly.FromDateTime(DateTime.Now), Time = TimeSpan.FromHours(10) }
+        };
+
+        _patientRepoMock
+            .Setup(repo => repo.GetPatientByIdAsync(patientId))
+            .ReturnsAsync(new Patient { Id = patientId });
+
+        _appointmentRepoMock
+            .Setup(repo => repo.GetAppointmentsBySpeciality(speciality, patientId))
+            .ReturnsAsync(appointments);
+
+        _mapperMock
+            .Setup(mapper => mapper.Map<List<AppointmentDto>>(appointments))
+            .Returns(new List<AppointmentDto>
+            {
+                new AppointmentDto { Date = DateOnly.FromDateTime(DateTime.Now), Time = TimeSpan.FromHours(10), Status = AppointmentStatuses.Scheduled }
+            });
+        
+        var result = await _appointmentService.GetAppointmentsBySpecialityId(speciality, patientId);
+        
+        Assert.True(result.Success);
+        Assert.NotNull(result.Data);
+        Assert.Single(result.Data);
```

Comment:
Use fluent assertions for all your tests. This is industry best practice assertion library